### PR TITLE
Initialize all values in the hrvo simulator, customize dock labels

### PR DESCRIPTION
**Pull request by jonathanlew**
### Description
as titled, should fix the sigfpe

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue

### CODE REVIEW

File: src/extlibs/hrvo/simulator.h [hunk 1]
```diff
@@ -260,16 +260,16 @@ class HRVOSimulator
     }
 
    private:
-    unsigned int frame = 0;
+    unsigned int frame;
```
Comment (nimazareian) [inline src/extlibs/hrvo/simulator.h]: Frame can actually be removed as its not used anymore

File: src/extlibs/hrvo/simulator.cpp [hunk 2]
```diff
@@ -45,11 +45,19 @@
...
       robot_constants(robot_constants),
+      global_time(0.0f),
+      time_step(time_step),
+      last_time_velocity_updated(0.0f),
       reached_goals(false),
-      kd_tree(std::make_unique<KdTree>(this))
+      kd_tree(std::make_unique<KdTree>(this)),
+      agents(),
+      friendly_robot_id_map(),
+      enemy_robot_id_map()
```
Comment (nimazareian) [inline src/extlibs/hrvo/simulator.cpp:59]: If we want to initialise everything, we should probably add ~~`world` and~~ `primitive_set` aswell.

**Review by nimazareian — CHANGES_REQUESTED** [review]

File: src/extlibs/hrvo/simulator.cpp (hunk 2 above)
Comment (jonathanlew) [reply src/extlibs/hrvo/simulator.cpp:59]: it should be there...

File: src/software/thunderscope/thunderscope.py [hunk 3]
```diff
@@ -68,6 +68,66 @@
 REFRESH_INTERVAL_MS = 5
 GAME_CONTROLLER_URL = "http://localhost:8081"
 
+from pyqtgraph.dockarea.Dock import DockLabel
+
+
+def updateStylePatched(self):
```
Comment (akhilveeraghanta) [inline src/software/thunderscope/thunderscope.py]: can you move this function outside of this file, put it in `dock_label_style.py` in thunderscope/ and import it here

also you can't really tell the difference between a tab and a dock (you can't drag tabs but you can drag docks which the current style differentiates pretty well imo and I personally prefer that). if you want it to be blue, can you just change the purple to blue?

**Review by nimazareian — APPROVED** [review]
Thanks for fixing this bug! :lady_beetle:

**Review by akhilveeraghanta — APPROVED** [review]
looks good!

**Pull request merged**